source/language_service/src/wrapper_refactor.rs

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Wrapper refactor code action logic: generates a zero-parameter wrapper operation
+// that supplies default / placeholder values for an existing operation's parameters.
+
+#[cfg(test)]
+mod tests;
+
+use qsc::hir::{
+    CallableKind, ItemKind, PatKind,
+    ty::{Prim, Ty},
+};
+use qsc::{
+    Span,
+    line_column::{Encoding, Range},
+};
+
+use crate::{
+    compilation::Compilation,
+    protocol::{CodeAction, CodeActionKind, TextEdit, WorkspaceEdit},
+};
+
+pub(crate) fn operation_refactors(
+    compilation: &Compilation,
+    source_name: &str,
+    span: Span,
+    encoding: Encoding,
+) -> Vec<CodeAction> {
+    let mut code_actions = Vec::new();
+    let user_unit = compilation.user_unit();
+    let package = &user_unit.package;
+    let source_map = &user_unit.sources;
+    let source = source_map
+        .find_by_name(source_name)
+        .expect("source should exist");
+    let source_span = compilation.package_span_of_source(source_name);
+
+    for (_, item) in package.items.iter() {
+        if !source_span.contains(item.span.lo) || span.intersection(&item.span).is_none() {
+            continue;
+        }
+        if let ItemKind::Callable(decl) = &item.kind {
```

Comment:
Does this catch lambdas too?

Reply:
It did, but I fixed that now so that lambdas with parameters won't trigger the code action.